src/main/java/com/machine_coding/snakes_and_ladders/services/impl/GameServiceImpl.java

```diff
@@ -0,0 +1,115 @@
+package com.machine_coding.snakes_and_ladders.services.impl;
+
+import com.machine_coding.snakes_and_ladders.entities.Board;
+import com.machine_coding.snakes_and_ladders.entities.Ladder;
+import com.machine_coding.snakes_and_ladders.entities.Player;
+import com.machine_coding.snakes_and_ladders.entities.Snake;
+import com.machine_coding.snakes_and_ladders.services.GameService;
+
+import java.util.Random;
+import java.util.Scanner;
+
+
+public class GameServiceImpl implements GameService {
+
+    private Board board;
+    private boolean playerReachDestination;
+    private Random random;
+
+    @Override
+    public int getPersonPoint() {
+        board.setPersonPoint((board.getPersonPoint() + 1) % board.getPlayers().length);
+        return board.getPersonPoint();
+    }
+
+    @Override
+    public void setUp() {
+
+        board = new Board(1);
+        Scanner scanner = new Scanner(System.in);
```

Comment:
Input/Output should be kept outside this service class. That should be in a driver class.

Reply:
Input moved to the main method of application.